README.md

```diff
@@ -0,0 +1,79 @@
+# pyLBBVAndPatch 
+
+A Lazy Basic Block Versioning + Copy and Patch JIT Interpreter for CPython.
+
+Python is a widely-used programming language. CPython is its reference implementation. Due to Python’s dynamic type semantics, CPython is generally unable to execute Python programs as fast as it potentially could with static type semantics.
+
+Last semester, while taking CS4215, we made progress on a technique for removing type checks and other overheads associated with dynamic languages known as [Lazy Basic Block Versioning (LBBV)](https://arxiv.org/abs/1411.0352). This work will be referred to as PyLBBV. More details can be found in our [technical report](https://github.com/pylbbv/pylbbv/blob/pylbbv/report/CPython_Tier_2_LBBV_Report_For_Repo.pdf). For an introduction to PyLBBV, refer to our [presentation](https://docs.google.com/presentation/d/e/2PACX-1vQ9eUaAdAgU0uFbEkyBbptcLZ4dpdRP-Smg1V499eogiwlWa61EMYVZfNEXg0xNaQvlmdNIn_07HItn/pub?start=false&loop=false&delayms=60000). 
+
+This orbital, we intend to refine PyLBBV. These include but are not limited to:
+- General refactoring
+- Bug fixing
+- Better unboxing + support unboxing of other types
+- More type specialised code
+
+Furthermore, we intend to further PyLBBV by implementing a [Copy and Patch JIT](https://arxiv.org/abs/2011.13127) on top of the type specialisation PyLBBV provides. The culmination of these efforts will allow further optimisations to be implemented. We hope that this effort can allow Python to be as fast as a statically typed language. PyLBBV's type propagator has been [submitted](https://github.com/faster-cpython/ideas/issues/564) for consideration to the Faster CPython team at Microsoft with positive response. Our work here will be made publically available so that it will benefit CPython and its users, and we plan to collaborate closely with the developers of CPython in the course of this project.
```

Comment:
not really implementing as we will use an external library, lemme try rephrase

```suggestion
Furthermore, we intend to further pyLBBV by integrating a [Copy and Patch JIT](https://arxiv.org/abs/2011.13127) (using code written externally by Brandt Bucher) on top of the type specialisation PyLBBV provides. The culmination of these efforts will allow further optimisations to be implemented. We hope that this effort can allow Python to be as fast as a statically typed language. Our work here will be made publically available so that it will benefit CPython and its users, and we plan to collaborate closely with the developers of CPython in the course of this project.
```